components/suggestion/switch_channel_provider.jsx

```diff
@@ -200,6 +216,35 @@ export default class SwitchChannelProvider extends Provider {
         this.formatChannelsAndDispatch(channelPrefix, suggestionId, channels, users);
     }
 
+    userWrappedChannel(user) {
+        let displayName = `@${user.username}`;
+
+        if ((user.first_name || user.last_name) && user.nickname) {
+            displayName += ` - ${Utils.getFullName(user)} (${user.nickname})`;
+        } else if (user.nickname) {
+            displayName += ` - (${user.nickname})`;
+        } else if (user.first_name || user.last_name) {
+            displayName += ` - ${Utils.getFullName(user)}`;
+        }
```

Comment:
LOC 220-228 We have several similar instances of this block. It might be better to have a utility function (`utils/utils.jsx`) and refer into it.
- https://github.com/mattermost/mattermost-webapp/blob/master/components/suggestion/at_mention_provider.jsx#L53-L61
- https://github.com/mattermost/mattermost-webapp/blob/master/components/suggestion/switch_channel_provider.jsx#L262-L274

Reply:
@saturninoabril these three places do have the same code but it's not the same as the `displayName` used in `switch_channel_provider.jsx`:
https://github.com/mattermost/mattermost-webapp/blob/master/components/suggestion/search_user_provider.jsx#L26-L32
https://github.com/mattermost/mattermost-webapp/blob/master/components/suggestion/at_mention_provider.jsx#L55-L61
https://github.com/mattermost/mattermost-webapp/blob/master/components/admin_console/user_autocomplete_setting.jsx#L30-L36

Reply:
Right but I think the we can move up `user.id === currentId`, like so:
```
if (completedChannels[user.id] || user.id === currentId) {
    continue;
}

Reply:
https://github.com/mattermost/mattermost-webapp/blob/master/components/suggestion/switch_channel_provider.jsx#L264-L266 `switch_channel_provider.jsx` - Right but I think we can move up `user.id === currentId`, like so:
```
if (completedChannels[user.id] || user.id === currentId) {
    continue;
}
```

Reply:
@saturninoabril the code is different. In `switch_channel_provider.jsx` it concatenates all the values to form the `displayName`. In the other three files it replaces the value based on the ifs.
`displayName += ` - ${Utils.getFullName(user)} (${user.nickname})`;`

Reply:
Ah I see, you're right :). Didn't notice that on quick look.